charts/thoras/templates/agent/daemonset.yaml

```diff
@@ -0,0 +1,56 @@
+{{- if .Values.thorasAgent.enabled }}
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: thoras-agent
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app.kubernetes.io/name: {{ .Chart.Name }}
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+spec:
+  selector:
+    matchLabels:
+      app: thoras-agent
+  template:
+    metadata:
+      labels:
+        app: thoras-agent
+        app.kubernetes.io/name: {{ .Chart.Name }}
+        helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+        app.kubernetes.io/managed-by: {{ .Release.Service }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+      {{- with .Values.thorasAgent.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+    spec:
+      serviceAccountName: thoras-collector
```

Comment:
It seems we only have a couple service accounts. The `thoras-collector` is used by several other deployments and I figured I'd reuse it as well. We can add an additional, agent-specific service if we feel that reuse isn't appropriate.

Reply:
Do you know what api calls the agent will need to make against the k8s api?

Reply:
I don't think it will be making any calls, for now. But this was needed to get access to `thoras-secret-registry` so the images could be pulled.